include/dr/shp/copy.hpp

```diff
@@ -13,20 +13,22 @@
 
 namespace shp {
 
+template <typename Src, typename Dest>
+concept is_syclmemcopyable = std::is_same_v<std::remove_const_t<Src>, Dest> &&
```

Comment:
I recommend putting this inside an anonymous namespace.

```cpp
namespace {
}
```

Reply:
We should not be defining any symbols except for `inline` symbols, but I'm also fine with `__detail`.

Reply:
> We should not be defining any symbols except for `inline` symbols, but I'm also fine with `__detail`.

Duplication can also come from putting classes in anonymous namespaces. The implementations of the methods will not be shared across compilation units.